pkg/runner_manager/manager.go

```diff
@@ -240,15 +252,85 @@ func (m *Manager) requestStreamVersion(ctx context.Context, s model.Stream, clie
 	default:
 		return nil, fmt.Errorf("invalid stream version %v", version)
 	}
+
+	outputOptions := "-c:v libx264 -preset veryfast -tune zerolatency -c:a aac -ar 44100 -b:a 128k -g 60 -bufsize 5000k -maxrate 5000k -b:v 3500k"
+	if !s.IsSelfStream() {
+		switch lh.StreamProtocol {
+		case model.RTSP:
+			outputOptions = "-c:a copy -c:v copy -rtsp_transport tcp -preset veryfast -tune zerolatency"
+		case model.SRT:
+			outputOptions = "-c:a copy -c:v copy -preset veryfast -tune zerolatency"
+		}
+	}
+
+	var input string
+	if !s.IsSelfStream() {
+		switch lh.StreamProtocol {
+		case model.RTSP:
+			input = fmt.Sprintf("rtsp://%s", ip)
+		case model.SRT:
+			input = fmt.Sprintf("srt://%s", ip)
+		default:
+			return nil, fmt.Errorf("invalid stream protocol %v", lh.StreamProtocol)
+		}
+	} else {
+		// Lecture is Selfstream
+		course, err := m.dao.CoursesDao.GetCourseById(ctx, s.CourseID)
+		if err != nil {
+			return nil, err
+		}
+		input = fmt.Sprintf("rtmp://localhost/%s-%d", course.Slug, s.ID)
+	}
 	return client.RequestStream(ctx, &protobuf.StreamRequest{
 		StreamId:            ptr.Take(uint64(s.ID)),
 		Version:             ptr.Take(version),
 		End:                 timestamppb.New(s.End),
-		FfmpegOutputOptions: ptr.Take("-c:a copy -c:v copy"),
-		Input:               ptr.Take(fmt.Sprintf("srt://%s", ip)),
+		FfmpegOutputOptions: ptr.Take(outputOptions),
+		Input:               ptr.Take(input),
 	})
 }
 
+func (m *Manager) RequestSelfStream(ctx context.Context, request *protobuf.SelfStreamRequest) (*protobuf.SelfStreamResponse, error) {
+	if _, err := m.dao.RunnerDao.Get(ctx, *request.Hostname); err != nil {
+		logger.Warn("Could not find runner for selfstream", "hostname", *request.Hostname)
+		return nil, err
+	}
+
+	if *request.StreamKey == "" {
```

Comment:
This should work, right? Using the protoc generated getters is usually safer than dereferencing a potentially nil string.

```suggestion
	if request.GetStreamKey() == "" {
```